src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -132,22 +132,37 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
             PackageDependency packageDependency = default;
             if (packageReferenceArgs.NoVersion)
             {
-                var latestVersion = await GetLatestVersionAsync(originalPackageSpec, packageReferenceArgs.PackageId, packageReferenceArgs.Logger, packageReferenceArgs.Prerelease);
-
-                if (latestVersion == null)
+                if (originalPackageSpec.RestoreMetadata.CentralPackageVersionsEnabled)
+                {
+                    var centralVersion = originalPackageSpec
+                        .TargetFrameworks
+                        .Where(tf => tf.CentralPackageVersions.ContainsKey(packageReferenceArgs.PackageId))
+                        .Select(tf => tf.CentralPackageVersions[packageReferenceArgs.PackageId])
+                        .Max();
```

Comment:
`Max()` is likely unnecessary since CPM settings should specify a single version for a package. If multiple versions are defined, MSBuild typically selects the last one, ensuring a single version remains after evaluation. Please let me know if I am missing something.

Reply:
I don't know how MSBuild generate the dg. I see a collection, so I choose the max version because when a dependency has many reference it's the max that is restored.

I followed your guidance and I replaced `Max` by `FirstOrDefault`

Reply:
CentralPackageVersions is a dictionary, so it'll always have 1 result